test/support/generator/get-facility-activation-transaction-generator.ts

```diff
@@ -0,0 +1,132 @@
+import { ENUMS, PROPERTIES } from '@ukef/constants';
+import { BundleStatusEnum } from '@ukef/constants/enums/bundle-status';
+import { LenderTypeCodeEnum } from '@ukef/constants/enums/lender-type-code';
+import { AcbsPartyId, DateOnlyString, UkefId } from '@ukef/helpers';
+import { AcbsGetBundleInformationResponseDto } from '@ukef/modules/acbs/dto/acbs-get-bundle-information-response.dto';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { GetFacilityActivationTransactionResponseDto } from '@ukef/modules/facility-activation-transaction/dto/get-facility-activation-transaction-response.dto';
+
+import { AbstractGenerator } from './abstract-generator';
+import { RandomValueGenerator } from './random-value-generator';
+
+export class GetFacilityActivationTransactionGenerator extends AbstractGenerator<FacilityActivationTransactionValues, GenerateResult, GenerateOptions> {
+  constructor(protected readonly valueGenerator: RandomValueGenerator, protected readonly dateStringTransformations: DateStringTransformations) {
+    super(valueGenerator);
+  }
+
+  protected generateValues(): FacilityActivationTransactionValues {
+    // Numeric enums needs filter to get possible values.
+    const possibleBundleStatuses = Object.values(ENUMS.BUNDLE_STATUSES).filter((value) => !isNaN(Number(value)));
+    const possibleLenderTypes = Object.values(ENUMS.LENDER_TYPE_CODES);
+
+    return {
+      initialBundleStatusCode: possibleBundleStatuses[this.valueGenerator.integer({ min: 0, max: possibleBundleStatuses.length - 1 })] as number,
+      bundleStatusCode: this.valueGenerator.stringOfNumericCharacters({ maxLength: 2 }),
+      bundleStatusDesc: this.valueGenerator.string({ maxLength: 20 }),
+      initiatingUserName: this.valueGenerator.string({ maxLength: 60 }),
+      accountOwnerIdentifier: this.valueGenerator.acbsPartyId(), // TODO: is this right to use acbsPartyId?
```

Comment:
@avaitonis

Reply:
Same as below please.

Reply:
Assigned ticket number to TODO @abhi-markan